### PR TITLE
feat: add install_component field

### LINKED ISSUE #103: Feature request: setup-gcloud should support installing optional components
### TL;DR

The `setup-gcloud` action should support installing optional components. Currently, the only way to install optional components is to add a `run` action to your `ci.yml` like:

```yaml
steps:
  - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
  - name: Install Google Cloud SDK components
    run: yes | gcloud components install beta cloud-firestore-emulator
```

In practice, this approach seems to take a long time to execute, and often flakes.

### LINKED ISSUE #330: Support gcloud alpha / beta
Currently, I have to run `gcloud components update` in order to run `gcloud alpha` and `gcloud beta` commands. Maybe we could add a flag to automatically get those ?

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #330
Fixes #103

### CODE REVIEW

File: .github/workflows/setup-gcloud-it.yml [hunk 1]
```diff
@@ -136,3 +136,35 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - id: build
+        name: Build dist
+        run: |-
+          npm install
```
Comment (bharathkkb) [inline .github/workflows/setup-gcloud-it.yml:153]:
```suggestion
          npm ci
```

File: src/setup-gcloud.ts [hunk 2]
```diff
@@ -17,12 +17,14 @@
...
+  authenticateGcloudSDK,
   getLatestGcloudSDKVersion,
-  isInstalled,
+  installComponent,
   installGcloudSDK,
-  setProject,
-  authenticateGcloudSDK,
+  isInstalled,
   parseServiceAccountKey,
+  runCmdWithJsonFormat,
```
Comment (bharathkkb) [inline src/setup-gcloud.ts:26]: Looks like this is unused

File: src/setup-gcloud.ts [hunk 3]
```diff
@@ -47,6 +49,15 @@ export async function run(): Promise<void> {
       core.addPath(path.join(toolPath, 'bin'));
     }
 
+    // Install additional components
+    const components = core.getInput('install_components');
+    if (components) {
+      await installComponent(
+        components.split(',').map((comp) => comp.trim()),
+        false,
```
Comment (bharathkkb) [inline src/setup-gcloud.ts:57]: Any reason why we are setting silent false explicitly? I noticed for other setupGcloudSDK commands we default to true.

**Review by bharathkkb — APPROVED** [review]
Lets also update [the readme inputs](https://github.com/google-github-actions/setup-gcloud#inputs)

File: .github/workflows/setup-gcloud-it.yml [hunk 4]
```diff
@@ -136,3 +136,35 @@ jobs:
...
+  components:
+    if: ${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name && github.actor != 'dependabot[bot]' }}
+    name: setup-gcloud with components
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v2
```
Comment (sethvargo) [inline .github/workflows/setup-gcloud-it.yml:149]: Do you need `actions/setup-node` here too?

File: src/setup-gcloud.ts (hunk 3 above)
Comment (averikitsch) [reply src/setup-gcloud.ts:57]: This was for debugging. silent = false will print command output to the console.

File: .github/workflows/setup-gcloud-it.yml [hunk 5]
```diff
@@ -123,6 +136,30 @@ jobs:
     if: ${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name && github.actor != 'dependabot[bot]' }}
     name: setup-gcloud with configured project
     runs-on: ${{ matrix.operating-system }}
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+          - uses: actions/checkout@v2
```
Comment (sethvargo) [inline .github/workflows/setup-gcloud-it.yml:144]: I think this might need to be outdented one or two?

File: README.md [hunk 6]
```diff
@@ -78,6 +78,8 @@ jobs:
     project ID for commands. Individual commands can still override the project
     with the `--project` flag. If unspecified, the action attempts to find the
     "best" project ID by looking at other inputs and environment variables.
+    
+- `isntall_components`: (Optional) List of Cloud SDK components to install
```
Comment (sethvargo) [inline README.md:82]: Typo

**Pull request merged**